packages/commonwealth/shared/schemas/createCommunitySchema.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  ChainBase,
+  ChainCategoryType,
+  ChainNetwork,
+  ChainType,
+} from 'common-common/src/types';
+import { z } from 'zod';
+import { ALL_COMMUNITIES } from '../../server/middleware/databaseValidationService';
+import { getFileSizeBytes } from '../../server/util/getFilesSizeBytes';
+
+export const MAX_COMMUNITY_IMAGE_SIZE_KB = 500;
+
+async function checkIconSize(val, ctx) {
+  const fileSizeBytes = await getFileSizeBytes(val);
+  if (fileSizeBytes === 0) {
+    ctx.addIssue({
+      code: z.ZodIssueCode.custom,
+      message: "Image url provided doesn't exist",
+    });
+  }
+  if (fileSizeBytes >= MAX_COMMUNITY_IMAGE_SIZE_KB * 1024) {
+    ctx.addIssue({
+      code: z.ZodIssueCode.custom,
+      message: `Image must be smaller than ${MAX_COMMUNITY_IMAGE_SIZE_KB}kb`,
+    });
+  }
+}
+
+export const createCommunitySchema = z.object({
+  id: z.string(),
+  name: z
+    .string()
+    .max(255)
+    .refine((data) => !data.includes(ALL_COMMUNITIES), {
+      message: `String must not contain '${ALL_COMMUNITIES}'`,
+    }),
+  chain_node_id: z.number().optional(), // corresponds to the chain field
+  community_namespace: z
+    .object({
+      namespace: z.string().optional(),
+      tx_receipt: z.any().optional(), // this comes from NamespaceFactory#deployNamespace
```

Comment:
@ianrowan Do we know the type of the tx_receipt? In the NamespaceFactory#deployNamespace it is any, but narrowing it down would help validation.

Is it of type TransactionReceipt from web3-core?
```
export interface TransactionReceipt {
    status: boolean;
    transactionHash: string;
    transactionIndex: number;
    blockHash: string;
    blockNumber: number;
    from: string;
    to: string;
    contractAddress?: string;
    cumulativeGasUsed: number;
    gasUsed: number;
    effectiveGasPrice: number;
    logs: Log[];
    logsBloom: string;
    events?: {
        [eventName: string]: EventLog;
    };
}
```

Reply:
That should be the appropriate type for tx_receipt, when we integrate the frontend I'll do another pass to solidify types, etc in those helper files. They're mainly working stubs atm so still need to be finalized